### DISCUSSION TITLE
deoendency injection

### DISCUSSION THREAD

**Discussion by wordcoolframework** [goravel/goravel#527] [General]
how use Repository pattern and or arcitectures for example : SOA and How can we use it in routing?

func Api() {
	userController := controllers.NewUserController()
	facades.Route().Get("/users/{id}", userController.Show)

	categoryRepository := *repositories.NewCategoryRepo() contracts.CategoryContract
	categoryService := *services.NewCategoryService(categoryRepository)
	categoryContrller := controllers.NewCategoryController(categoryService)
	facades.Route().Get("api/categories", categoryContrller.Index)
}

**Comment by hwbrzzl** [comment]
Sorry, didn't get you, could you describe it more?
